### PR TITLE
New nametag.js

### LINKED ISSUE #34: nametags app: Redesign the application to use the entity Billboard mode
The current application "nametags" is quite old and instead to use the entity's billboardMode it do it by adjusting the orientation in a process. (billboardMode was probably not implemented when it has been coded.)

This app is good a complete revision. (it might have other issues or maybe use some deprecated things).

### PR DESCRIPTION
**Pull request by Armored-Dragon**
New stylized nametag.js for Overte with proper billboarding.
Testing would be appreciated before merging.

Resolves #34

### CODE REVIEW

**Comment by Zetaphor** [comment]
This is a big improvement over the current implementation!

image

**Comment by AnotherFoxGuy** [comment]
It looks good, There is only one thing that I've noticed is that it still has the issue of being inside the avatar sometimes 

image

**Comment by Zetaphor** [comment]
I had the same issue with my avatar due to my height and the hat

Is it possible to calculate an avatars bounding box to adjust this?

**Comment by Armored-Dragon** [comment]
> I had the same issue with my avatar due to my height and the hat
> 
> Is it possible to calculate an avatars bounding box to adjust this?

Currently the nametags are hardcoded to show up roughly where an avatar head *should* be. I'm looking into ways to address this.

**Comment by vegaslon** [comment]
yes sadly not all avatars have a HeadTop bone to tell you where the top of the head is.

**Comment by Armored-Dragon** [comment]
> yes sadly not all avatars have a HeadTop bone to tell you where the top of the head is.

That's what I've found so far. If anyone has any other ideas on how to provide a consistant nametag position based on the head, feel free to suggest.

**Comment by AnotherFoxGuy** [comment]
> That's what I've found so far. If anyone has any other ideas on how to provide a consistant nametag position based on the head, feel free to suggest.

The old version of the nametags just grabs the head joint and bases the nametag position on that
https://github.com/overte-org/overte/blob/master/scripts/simplifiedUI/ui/simplifiedNametag/resources/modules/nameTagListManager.js#L376-L377

File: applications/nametags/nametags.js [hunk 1]
```diff
@@ -1,381 +1,154 @@
...
-                button.clicked.connect(onButtonClicked);
-            }
-        }, 2500);
-
-        setVisible(visible);  // Starts updating if visible.
+//
+// Copyright 2023 Overte e.V.
+//
+// Written by Armored Dragon
+
```
Comment (ksuprynowicz) [inline applications/nametags/nametags.js:7]: Apache 2.0 license information is missing
Check out the header here:
https://github.com/overte-org/community-apps/blob/master/applications/aaswitcher/app-aaSwitcher.js

**Review by ksuprynowicz — APPROVED** [review]
everything looks fine :)

**Comment by ksuprynowicz** [comment]
@Armored-Dragon Is it ready for merging in your opinion?

**Comment by Armored-Dragon** [comment]
> @Armored-Dragon Is it ready for merging in your opinion?

Yes. As long as there are no more remaining issues or oversights that anyone had noticed, I think this is now ready.

**Pull request merged**